src/rpcblockchain.cpp

```diff
@@ -665,6 +666,14 @@ UniValue getblockchaininfo(const UniValue& params, bool fHelp)
             "        },\n"
             "        \"reject\": { ... }      (object) progress toward rejecting pre-softfork blocks (same fields as \"enforce\")\n"
             "     }, ...\n"
+            "  ],\n"
+            "  \"upgrades\": [            (array) status of non-forward compatible upgrades\n"
```

Comment:
Are we sure we mean "forward-compatible" as opposed to "backward-compatible" here? I don't know what precise meaning we're ascribing to those terms.

Reply:
Yes, "status of upgrades" is sufficient IMO.

Reply:
Ok, will go with daira's suggestions, since this might be used for non-forking upgrades in the future. Should there be some format to the upgradeid that indicates it's non-backwards compatible?